daemon: factor out iptables code into pkg/iptables

ianvernon opened the pull request:
Navigating daemon.go is confusing; there is a lot of code in various places that
could be separated out into smaller files that more logically indicate the
function of the code contained inside of them. Move iptables related code to its
own file accordingly.

Signed-off by: Ian Vernon <ian@cilium.io>

No functional changes are intended in this PR.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5440)
<!-- Reviewable:end -->

vadorovsky reviewed (commented):
Maybe we should have a separate package for that (i.e. `pkg/firewall`)? Then in the future it will be easier to abstract that functionality and provide support for nftables and firewalld.

aanm reviewed (dismissed):
Waiting for addressing Thomas comment

Commit:
daemon: move iptables-related code to pkg/iptables

Navigating daemon.go is confusing; there is a lot of code in various places that
could be separated out into smaller files / packages that more logically
indicate the function of the code contained inside of them. Move iptables
related code to its own package, `pkg/iptables`, accordingly.

Signed-off by: Ian Vernon <ian@cilium.io>

ianvernon commented on pkg/datapath/firewall/iptables.go:
```diff
@@ -0,0 +1,346 @@
+	}
+}
+
+func InstallIptablesRules() error {
```
Fixed in next upload.

ianvernon commented on pkg/datapath/firewall/iptables.go:
```diff
@@ -0,0 +1,346 @@
+	// Masquerade specifies whether or not to masquerade packets from endpoints
+	// leaving the host
+	Masquerade                bool
+	PrependIptablesChainsName bool
```
Removed in next upload.

ianvernon commented on pkg/datapath/firewall/doc.go:
```diff
@@ -0,0 +1,16 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package firewall manages non-BPF firewall solutions (e.g., iptables).
```
Fixed in next upload.

ianvernon commented on pkg/datapath/firewall/iptables.go:
```diff
@@ -0,0 +1,346 @@
+	},
+}
+
+func RemoveIptablesRules() {
```
Fixed in next upload.

joestringer commented on daemon/daemon.go:
```diff
@@ -788,8 +476,8 @@ func (d *Daemon) compileBase() error {
-		d.removeIptablesRules()
-		if err := d.installIptablesRules(); err != nil {
+		iptables.RemoveIptablesRules()
+		if err := iptables.InstallIptablesRules(); err != nil {
```
iptables.xxxxIptables stutters.

ianvernon replied:
Fixed in next upload.

joestringer reviewed (approved):
Minor stutter in the function names, but that's a real nit. LGTM.

aanm commented on pkg/defaults/defaults.go:
```diff
@@ -94,4 +94,7 @@ const (
 	IdentityChangeGracePeriod = 25 * time.Second
+
+	// ExecTimeout is the execution timeout to use in init.sh executions
+	ExecTimeout = 300 * time.Second
```
This variable should have a different name as it's only used in `init.sh`

ianvernon replied:
It's used in various places - the comment was out-of-date. The comment is fixed in the next upload.
